src/raven.js

```diff
@@ -498,6 +496,41 @@ Raven.prototype = {
         }
     },
 
+    showReportDialog: function (options) {
+        if (!window.document) // doesn't work without a document (React native)
+            return;
+
+        options = options || {};
+
+        var lastEventId = options.eventId || this.lastEventId();
+        if (!lastEventId) {
+            throw new RavenConfigError('Missing eventId');
```

Comment:
I think it's really hard to get into an infinite loop situation here, because when calling `showReportDialog`:

1) If you don't have a DSN, then Raven wasn't installed and the global error handler isn't present – the exception is thrown but nothing catches it and control flow ends.

2) If you don't have an `eventId` (or there was no `lastEventId`), then Raven will throw an exception, and this will be caught by the global error handler. Now if `showReportDialog` is called a second time, now there **is** a `lastEventId` and the function will no longer error.

Also, if you have an `onerror` handler, and inside that handler you throw *another error* (e.g. `RavenConfigError`), `onerror` doesn't get called again in an infinite loop – it just stops there:

http://jsbin.com/qoqoviduru/1/edit?html,js

The user would have to use a `setTimeout` call to leave the `onerror` stack to reach this infinite loop scenario:

http://jsbin.com/moxulabigi/1/edit?html,js,output *(warning, this will freeze your browser)*

But I mean, if they did this, any kind of bug in their code would trigger the infinite/setTimeout scenario – not strictly this `RavenConfigError`. So altogether, I think this scenario is really unlikely, and would be their fault anyways.

I'm going to move ahead with merging the branch. It's flagged experimental anyways.